docs/en/week09/09-3.md

```diff
@@ -0,0 +1,387 @@
+```
+
+We use log variance instead of varaince (change of scale) because we want to ensure
+1. Variance is non-negative.
+2. Full range of variance, to make training stable.
+
+Recall the free energy for VAE, 
+$$
+    F(y)=C(y,\tilde{y})+\beta D_{KL}[\mathcal{N}(\mu,v)||\mathcal{N}(0,1)]
+        =C(y,\tilde{y})+\frac{\beta}{2}\sum_{i=1}^{d}v_i-log(v_i)-1+\mu_i^2
+$$
+
+To regualrize the expressivity of the latent, we include KL-divergence between the Gaussian of latent variable and a Normal distribution (\mathcal{N}(0,1)). $C(y,\hat{y})$ represents the reconstruction term. (Also see [Week 8-Practicum]({{site.baseurl}}/images/week8/08-3) for bubble explaination of VAE loss)
+
+Therefore, we define the loss function as 
+```
+def loss_function(x_hat, x, mu, logvar):
+    BCE = nn.functional.binary_cross_entropy(
+        x_hat, x.view(-1, 784), reduction='sum'
+    )
+    KLD = 0.5 * torch.sum(logvar.exp() - logvar - 1 + mu.pow(2))
+
+    return BCE + KLD
+```
+
+Since, VAE is a generative model, we sample from the distribution to generate the following digits:
+
+```
+N=16
+z=torch.rand((N,d)).to(device)
+sample=model.decoder(z)
+```
+
+<center>
+<img src="{{site.baseurl}}/images/week09/09-3/fig_10_cluster_samples.png" height="300px" style="background-color:#DCDCDC;"/><br>
+<b>Fig. 8</b>: TSNE visualization of samples generated through VAE.
+</center>
+
+The regions (classes) get segregated as the reconstruction term forces the latent space to get well defined. The data gets clustered into classes without actually using the labels.
+
+### Generative Adversarial Nets (GANs)
+
+GANs have the same feel as DAE with some tweaks. DAE involves generating corrput samples using the input and a distribution followed by denoising them using a decoder. GANs instead directly sample the distribution (without the input) to produce output $\hat{y}$ using the Generator (or decoder in DAE terms). The input $y$ and $\hat{y}$ are provided to the Cost network separately to measure incomptability between them.
+
+<center>
+<img src="{{site.baseurl}}/images/week09/09-3/fig_11_gan_vs_dae.png" height="400px" style="background-color:#DCDCDC;"/><br>
+<b>Fig. 9</b>: Comparision of GANs(left) and DAE(right) architectures.
+</center>
+
+Similarly, we can extend the analogy with VAE. In contrast to VAE where the sampler is conditioned on the output of the encoder, in GANs there is an unconditioned sampler and again, decoder corresponds to Generator.
+
+<center>
+<img src="{{site.baseurl}}/images/week09/09-3/fig_12_gan_vs_vae.png" height="400px" style="background-color:#DCDCDC;"/><br>
+<b>Fig. 10</b>: Comparision of VAE(left) and GANs (right) architectures.
+</center>
+
+The Generator maps the latent space to data space:
+$$
+G:\mathcal{Z}\rightarrow \mathbb{R}^n,z \rightarrow \hat{y}
+$$.
+The observed $y$ and $\hat{y}$ are fed to a Cost network to measure incompatibility:
+$$
+C:\mathcal{R}^n\rightarrow R, y \vee \hat{y} \rightarrow c
+$$
```

Comment:
You need an extra empty line before and after each equation.
Read [the instructions](https://atcold.github.io/pytorch-Deep-Learning/en/about/).

Reply:
All vectors should use `\vect{}`.

Reply:
Corrected

Reply:
The output of the corruption module in a DAE is `\hat{y}` as well.
The hat means it's a contrastive sample.
Both DAE and GAN are contrastive techniques, where we explicitly push up (the hat looks like an upwards arrow) on the contrastive samples.
The prediction for y is `\tilde{y}`, which means ~y, or *circa* y (latin for “almost“).

But I see I failed to explain this in class too clearly.
Next time around I'll make sure to put a legend somewhere.
Actually…

![image](https://user-images.githubusercontent.com/2119355/119197801-5327c080-ba56-11eb-87a9-db46ad869339.png)

Reply:
I guess I need to add an additional input (output?) `\hat{y}`.